internal/store/build_result.go

```diff
@@ -270,9 +270,18 @@ type BuildState struct {
 	// This must be liberal: it's ok if this has too many files, but not ok if it has too few.
 	FilesChangedSet map[string]bool
 
-	// Whether there was a manual trigger
-	// TODO(nick): Replace this with a BuildReason
-	NeedsForceUpdate bool
+	// There are two kinds of triggers:
+	//
+	// 1) If a resource is in trigger_mode=TRIGGER_MODE_MANUAL, then the resource doesn't
+	//    build until there's an explicit trigger. Once we have an explicit trigger,
+	//    we can do a normal image build or normal live_update.
+	//
+	// 2) If a resource is in trigger_mode=TRIGGER_MODE_AUTO, then the resource auto-builds.
```

Comment:
not quite--as stated above, you can still force a full build on a manual resource. This maybe belies the naming change behind this PR, let's talk in the high-level comments.